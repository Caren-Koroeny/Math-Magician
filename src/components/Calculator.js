import Display from './Display';
import NumbersKeys from './NumbersKeys';
import Operators from './Operators';

const Calculator = () => (
  <div className="mainContainer">
    {/* add child Components */}
    <Display />
    <NumbersKeys />
    <Operators />

  </div>
);

export default Calculator;
