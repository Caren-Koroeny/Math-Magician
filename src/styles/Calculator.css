@import url('https://fonts.googleapis.com/css2?family=Crete+Round&family=Inter:wght@400;500;600;700;800&family=Poppins&family=Roboto:wght@700&display=swap');

* {
  padding: 0;
  margin: 0;
  box-sizing: 0;
  font-family: "Poppins", sans-serif;
}

/* App.js */
.calculatorMain-container {
  display: flex;
  align-items: center;
  justify-content: center;
  height: 100vh;
}

/* calculator.js */
.mainContainer {
  width: 90%;
  margin: 0 auto;
  display: grid;
  grid-template-columns: repeat(4, minmax(0, 1fr));
  grid-template-rows: repeat(6, minmax(0, 1fr));
  max-inline-size: 22rem;
  border-radius: 10px;
  overflow: hidden;
  box-shadow:
    0 3px 6px 0 rgba(0, 0, 0, 0.6)
    0 2px 6px 0 rgba(0, 0, 0, 0.6);
}

/* Display.js */
.displayscreen-count-section {
  grid-column: 1/ 5;
  grid-row: 1 / 2;
}

.input-screen {
  width: 100%;
  height: 100%;
  padding: 15px 5px;
  background: rgb(122, 130, 136) none repeat scroll 0% 0%;
  color: #fff;
  border: none;
  outline: none;
  text-align: end;
}

/* NumbersKey.js */
.allNumbers {
  grid-column: 1/ 4;
  grid-row: 3/ 7;
  display: grid;
  grid-template-columns: repeat(3, minmax(0, 1fr));
}

.btn {
  background-color: #e0e0e0;
  padding: 10px 0;
  border: 0.01px solid hsl(0, 0%, 80%);
  outline: none;
  transition: 0.05s ease-in-out;
  font-size: 1.1rem;
}

.btn:active {
  background-color: hsl(0, 0%, 78%);
}

.btn-top {
  grid-row: 2 / 3;
}

.btn-side {
  background-color: #f5913e;
  grid-column: 4/ 5;
}

.btn-side:active {
  background-color: hsl(27, 90%, 50%);
}

.col-span-2 {
  grid-column: 1 / 3;
}

