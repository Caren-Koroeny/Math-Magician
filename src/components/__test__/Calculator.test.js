import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from '../Calculator';

it('renders Calculator correctly', () => {
  const tree = renderer.create(<Calculator />);
  expect(tree).toMatchSnapshot();
});