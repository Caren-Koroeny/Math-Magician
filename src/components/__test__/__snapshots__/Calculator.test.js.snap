// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Calculator correctly 1`] = `
<div
  className="mainContainer"
>
  <div
    className="displayscreen-count-section"
  >
    <span
      className="input-screen"
    >
      0
    </span>
  </div>
  <div
    className="allNumbers"
  >
    <button
      className="btn btn-primary"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="btn btn-primary"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="btn btn-primary"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="btn btn-primary"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="btn btn-primary"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="btn btn-primary"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="btn btn-primary"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="btn btn-primary"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="btn btn-primary"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="btn btn-primary col-span-2"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="btn btn-primary"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
  </div>
  <button
    className="btn btn-top"
    onClick={[Function]}
    type="button"
  >
    AC
  </button>
  <button
    className="btn btn-top"
    onClick={[Function]}
    type="button"
  >
    +/-
  </button>
  <button
    className="btn btn-top"
    onClick={[Function]}
    type="button"
  >
    %
  </button>
  <button
    className="btn btn-side"
    onClick={[Function]}
    type="button"
  >
    รท
  </button>
  <button
    className="btn btn-side"
    onClick={[Function]}
    type="button"
  >
    x
  </button>
  <button
    className="btn btn-side"
    onClick={[Function]}
    type="button"
  >
    -
  </button>
  <button
    className="btn btn-side"
    onClick={[Function]}
    type="button"
  >
    +
  </button>
  <button
    className="btn btn-side"
    onClick={[Function]}
    type="button"
  >
    =
  </button>
</div>
`;
